db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ConcatenationOperatorNode.java

```diff
@@ -138,156 +138,67 @@ public ValueNode bindExpression(FromList fromList,
          * the other parameter, with maximum length for that type.
          */
 
-        if (getLeftOperand().requiresTypeFromContext()) {
-            if (getRightOperand().requiresTypeFromContext()) {
-                throw StandardException.newException(
-                        SQLState.LANG_BINARY_OPERANDS_BOTH_PARMS, operator);
-            }
-
-            TypeId leftType;
-
-            /*
-             * * A ? on the left gets its type from the right. There are eight *
-             * legal types for the concatenation operator: CHAR, VARCHAR, * LONG
-             * VARCHAR, CLOB, BIT, BIT VARYING, LONG BIT VARYING, and BLOB. * If
-             * the right type is BLOB, set the parameter type to BLOB with max
-             * length. * If the right type is one of the other bit types, set
-             * the parameter type to * BIT VARYING with maximum length. * * If
-             * the right type is CLOB, set parameter type to CLOB with max
-             * length. * If the right type is anything else, set it to VARCHAR
-             * with * maximum length. We count on the resolveConcatOperation
-             * method to * catch an illegal type. * * NOTE: When I added the
-             * long types, I could have changed the * resulting parameter types
-             * to LONG VARCHAR and LONG BIT VARYING, * but they were already
-             * VARCHAR and BIT VARYING, and it wasn't * clear to me what effect
-             * it would have to change it. - Jeff
-             */
-            if (getRightOperand().getTypeId().isBitTypeId()) {
-                if (getRightOperand().getTypeId().isBlobTypeId())
```

Comment:
how can this all be removed?

Reply:
This code is duplicated for left and right, i removed it all and refactored it below